game-core/src/main/java/games/strategy/triplea/settings/ProtectedStringClientSetting.java

```diff
@@ -0,0 +1,44 @@
+package games.strategy.triplea.settings;
+
+import java.util.Optional;
+
+import games.strategy.security.CredentialManager;
+import games.strategy.security.CredentialManagerException;
+
+class ProtectedStringClientSetting extends StringClientSetting {
+
+  ProtectedStringClientSetting(final String name, final String defaultValue) {
+    super(name, protect(defaultValue));
+  }
+
+  private static String protect(final String value) {
+    try (CredentialManager manager = CredentialManager.newInstance()) {
+      return manager.protect(value);
+    } catch (final CredentialManagerException e) {
+      throw new IllegalStateException("CredentialManager needs to be available in order to protect strings.", e);
+    }
+  }
+
+  private static String unprotect(final String encodedValue) {
+    try (CredentialManager manager = CredentialManager.newInstance()) {
+      return manager.unprotectToString(encodedValue);
+    } catch (final CredentialManagerException e) {
+      throw new IllegalStateException("CredentialManager needs to be available in order to unprotect strings.", e);
+    }
+  }
+
+  @Override
+  public void setValue(final String value) {
+    super.setValue(protect(value));
```

Comment:
Was it not possible to do this in `formatValue()` (and unprotect in `parseValue()`)?  That would keep the subclass variation in the abstract methods instead of the superclass having to be concerned with defining proper override semantics (via Javadocs) in `setValue()` and `getValue()`.

I haven't really thought about the default value stuff yet, and not sure if that's the reason you took this route.  However, one could argue that a "protected/secret" string probably wouldn't have a meaningful default value in the first place. :smile:

Reply:
If you look at the commit history, you'll see that this was actually my first attempt.
It worked as far as I could tell, but I was concerned this would keep the raw string in memory.
Looking back at the code it seems that we directly push the string to he preferences store, so that wouldn't be really an argument.
I'll check if this works fine in combination with the default value tomorrow, and revert it, if possible.